helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java

```diff
@@ -145,13 +146,49 @@ public void testPostRequestFormat() throws IOException {
     Assert.assertEquals(json.get("data").asText(), "{}");
   }
 
+  @Test
+  public void testGetPartitionsStoppableCheckWhenMultiThreads()
+      throws IOException, InterruptedException {
+    int parallelCount = 3;
+    MockCustomRestClient customRestClient = new MockCustomRestClient(_httpClient, parallelCount);
+    HttpResponse httpResponse = mock(HttpResponse.class);
+    StatusLine statusLine = mock(StatusLine.class);
+
+    when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_NOT_FOUND);
+    when(httpResponse.getStatusLine()).thenReturn(statusLine);
+    when(_httpClient.execute(any(HttpPost.class))).thenReturn(httpResponse);
+
+    Runnable runnable = () -> {
+      try {
+        LOG.info("Executing {}", Thread.currentThread().getName());
```

Comment:
Is logging really necessary for tests? Make it an in-line comment instead?

Reply:
It's for debug purpose, since each thread will have a different name.